learning.py

```diff
@@ -1049,14 +1049,35 @@ def grade_learner(predict, tests):
     return mean(int(predict(X) == y) for X, y in tests)
 
 
-def train_test_split(dataset, start, end):
-    """Reserve dataset.examples[start:end] for test; train on the remainder."""
-    start = int(start)
-    end = int(end)
-    examples = dataset.examples
-    train = examples[:start] + examples[end:]
-    val = examples[start:end]
-    return train, val
+def train_test_split(dataset, start = None, end = None, test_split = None):
+    """If you are giving 'start' and 'end' as a parameter,
+    then it will return testing set from index 'start' to 'end'
+    and rest for training.
+    If you give 'test_split' as parameter then it will first shuffle the
+    dataset then return test_split * 100% as testing set and rest as 
+    training set.
+    """
+    
+    if start == None and end != None:
```

Comment:
I believe these if statements are not needed, since we are trying to keep the code as close to pseudocode form as possible.